### ISSUE TITLE
Masking is being repeated in `masking` training mode

### ISSUE THREAD

**Issue by kacpnowak** [ecmwf/WeatherGenerator#267]
### What happened?

When evaluating model in `masking` training mode, each sample should be masked randomly. However the masks are reused by what looks like 7 or 8 consecutive samples. I attach gif for illustration. 

![Image](https://github.com/user-attachments/assets/e4d0e38d-334c-47ad-99b7-4634a1927bea)

### What are the steps to reproduce the bug?

Train model with training mode set to `masking` and evaluate it on data having more than 100 samples

### Version

0.1

### Platform (OS and architecture)

Levante Linux

### Relevant log output

```shell

```

### Organisation

AWI

**Comment by clessig** [comment]
@kacpnowak : Can you try this patch

```
--- a/src/weathergen/datasets/multi_stream_data_sampler.py
+++ b/src/weathergen/datasets/multi_stream_data_sampler.py
@@ -418,4 +418,7 @@ class MultiStreamDataSampler(torch.utils.data.IterableDataset):
                 + f" : dataset [{local_start},{local_end}) : [{iter_start},{iter_end})"
             )
 
+        # ensure the tokenizers use different seeds
+        self.tokenizer.reset()
+
         return iter_start, iter_end
 
--- a/src/weathergen/datasets/tokenizer_masking.py
+++ b/src/weathergen/datasets/tokenizer_masking.py
@@ -130,7 +130,9 @@ class TokenizerMasking:
         return self.size_time_embedding
 
     def reset(self) -> None:
-        self.rng = np.random.default_rng(int(time.time()))
+        worker_info = torch.utils.data.get_worker_info()
+        div_factor = worker_info.num_workers if worker_info is not None else 1
+        self.rng = np.random.default_rng(int(time.time() / div_factor))
 
     def batchify_source(
         self,
```

**Comment by clessig** [comment]
Yes, open a PR.

You mean the loss is increases by 5%? At the beginning of training?

**Comment by kacpnowak** [comment]
I mean the loss computed on the evaluation. I have it evaluated on one year of data to test MTM over entire seasonal cycle
